### PR TITLE
feat: compile lm_head using rbln backend

### PR DESCRIPTION
**Pull request by pei0033**
### 🚀 Summary of Changes

This PR adds compilation support for the `logits processor` (lm_head) using the `rbln` backend in the V1 engine. The logits processor is now compiled with `torch.compile` using the rbln backend alongside the main model, enabling optimized inference for the language model head component.

---

### 📌 Related Issues / Tickets

* Related to #92 

---

### ✅ Type of Change

* [x] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [x] 🧬 Core engine changes (`core`)
* [ ] 🛠 Bug fix (`bug-fix`)
* [x] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1.Run `USE_VLLM_MODEL=1  VLLM_USE_V1=1 python offline_inference_basic.py`

---

### 📋 Checklist

* [x] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [x] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

- This change specifically targets the V1 engine (rbln_model_runner.py)
- The logits_processor compilation uses the same compile context and options as the main model to ensure consistency
- Target branch: fix/v1_engine

---

### CODE REVIEW

File: vllm_rbln/v1/worker/rbln_model_runner.py [hunk 1]
```diff
@@ -1253,6 +1257,15 @@ def load_model(self) -> None:
 
             self.compile_context = CompileContext(use_weight_sharing=True)
             self.model_executable = self._compile_model(self.model)
+            self.model.logits_processor = torch.compile(
```
Comment (rebel-jiwoopark) [inline vllm_rbln/v1/worker/rbln_model_runner.py:1260]: It seems to be good to modify the code to consider `eager mode` and the `not COMPILE_MODEL` environment variable.
You might find it helpful to look for conditions like the following as a reference:
`enforce_eager`
`not envs.RBLN_COMPILE_MODEL`

Comment (rebel-jiwoopark) [reply vllm_rbln/v1/worker/rbln_model_runner.py:1260]: Okay, I understood. Thank you for your checking!

Comment (rebel-jiwoopark) [reply vllm_rbln/v1/worker/rbln_model_runner.py:1260]: I missed that this code is already under the corresponding conditional branch.
